src/main/java/org/apache/commons/text/similarity/SorensenDicesSimilarity.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.text.similarity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ *
+ * @since 1.7
+ */
+public class SorensenDicesSimilarity implements SimilarityScore<Double> {
+
+    /**
+     * @param left  the first CharSequence, must not be null
+     * @param right the second CharSequence, must not be null
+     * @return result similarity
+     * @throws IllegalArgumentException if either CharSequence input is {@code null}
+     */
+
+    @Override
+    public Double apply(final CharSequence left, final CharSequence right) {
+
+        if (left == null || right == null) {
+            throw new IllegalArgumentException("CharSequences must not be null");
+        }
+
+        if (left.equals(right)) {
+            return 1d;
+        }
+
+        if ("".equals(left) || "".equals(right)) {
+            return 0d;
+        }
+
+        Set<String> nLeft = new HashSet<String>();
+        Set<String> nRight = new HashSet<String>();
+
+        for (int i = 0; i < left.length() - 1; i++) {
+            char chr = left.charAt(i);
+            char nextChr = left.charAt(i + 1);
+            String bi = "" + chr + nextChr;
+            nLeft.add(bi);
+        }
+        for (int j = 0; j < right.length() - 1; j++) {
```

Comment:
Code duplication can be moved to a method:
```
private Set<String> createBigrams(CharSequence cs);
```
Note that in my general comment on the PR that a bigram (2 * 16-bit `char`s) can be represented as a 32-bit `int`. This would be faster for construction than `String` and faster for the hash set.

Reply:
Thanks for the suggestion, corrected code. looks cleaner now. let's see whats community expects. as this is limited to text and not binary similarity. like the idea of int thought